AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreConfiguration.swift

```diff
@@ -101,13 +105,15 @@ extension DataStoreConfiguration {
         },
         syncInterval: TimeInterval = DataStoreConfiguration.defaultSyncInterval,
         syncMaxRecords: UInt = DataStoreConfiguration.defaultSyncMaxRecords,
-        syncPageSize: UInt = DataStoreConfiguration.defaultSyncPageSize
+        syncPageSize: UInt = DataStoreConfiguration.defaultSyncPageSize,
+        syncExpressions: [DataStoreSyncExpression] = []
```

Comment:
nit: looks like we process a `DataStoreSyncExpression` per model, I wonder if changing this array to a dictionary `[ModelName: DataStoreSyncExpression]` makes this more explicit at the caller site (feel free to ignore if you've already considered it)

Reply:
I looked into doing this, but i think it can end up being more confusing for the customer.  For example, they are currently doing something like:
```
            let syncExpression1 = DataStoreSyncExpression.syncExpression(Todo.schema, where: {
                Todo.keys.tag == AppDelegate.tagToSearch
            })
            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(),
                                                     configuration: .custom(syncExpressions: [syncExpression1]))
```
If I understand you correctly, are you suggesting to do something the following?
```
            let syncExpression1 = DataStoreSyncExpression.syncExpression(Todo.schema, where: {
                Todo.keys.tag == AppDelegate.tagToSearch
            })
            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(),
                                                     configuration: .custom(syncExpressions: ["\(Todo.schema.name)":syncExpression1]))
```